### DISCUSSION TITLE
Connecting children nodes say n1 in parent1 to n2 in parent2 directly with ELKJS without connecting parent port

### DISCUSSION THREAD

**Discussion by ROYopedia** [clientIO/joint#2229] [Q&A]
### Introduction

![image](https://github.com/clientIO/joint/assets/22453886/e5fcf3df-1903-4cf8-a714-3484643fbf81)

I am analyzing elk with jointjs where I am modifying existing demo code and trying connecting two children nodes in different parents directly, but not via parent ports.

lets take and example from the given demo,

I want to connect "Sleep child node's south port of parent Interface - slow"  to " Channel2 parent's Display 's west port"

I tried changing the json file following way(part of the code).
```js
 "edges": [
 {
            "id": "InnerNodeConnection",
            "sources": [
              "SLOW_SOUTH"
            ],
            "targets": [
              "channel2_Display_WEST"
            ]
          }
]
```
Getting error in addEdges function in demo code
![image](https://github.com/clientIO/joint/assets/22453886/9ebbcaa4-ef3a-4abb-846c-5b2618916fe4)

Any help would be appreciable.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
It seems that the sample code is not 100% error-prone. The code of the demo needs to be changed so that edges are added after all nodes have been added.

Edges are currently added [here](https://github.com/clientIO/joint/blob/master/demo/elk/src/app.js#L98-L100). This can lead to a situation where the target node has not yet been added to the `mapPortIdToShapeId` structure.
